genshin/errors.py

```diff
@@ -166,15 +166,20 @@ class RedemptionClaimed(RedemptionException):
     -500004: VisitsTooFrequently,
     -502001: "User does not have this character.",
     -502002: "Calculator sync is not enabled.",
+    # diary
+    -400005: "Parameter error",
+    -501101: "This character has not reached Lv.10, please select another character.",
     # mixin
     -1: InternalDatabaseError,
     1009: AccountNotFound,
     # redemption
+    -1065: RedemptionInvalid,
     -1071: InvalidCookies,
     -1073: (AccountNotFound, "Account has no game account bound to it."),
     -2001: (RedemptionInvalid, "Redemption code has expired."),
     -2003: (RedemptionInvalid, "Redemption code is incorrectly formatted."),
     -2004: RedemptionInvalid,
+    -2014: (RedemptionInvalid, "Redemption code not activated"),
```

Comment:
If this is the exact message that the API already returns, you don't need to add it since it already gets detected as a `RedemptionException` in `raise_for_retcode`

Reply:
Oh do the messages change based on the language?

If so then I'll just switch to looking at the retcode range instead I suppose.

Reply:
All redemption exception messages are based on the language. Other genshin exceptions, some based on language, some not.